scalafix-core/src/main/scala/scalafix/rewrite/ExplicitReturnTypes.scala

```diff
@@ -58,14 +85,39 @@ case class ExplicitReturnTypes(mirror: Mirror)
         typ <- defnType(defn)
       } yield ctx.addRight(replace, s": ${typ.treeSyntax}")
     }.to[Seq]
+
+    import ctx.config.explicitReturnTypes._
+
+    def matchesMemberVisibility(mods: Seq[Mod]): Boolean =
+      memberVisibility.contains(visibility(mods))
+
+    def matchesMemberKind(defn: Defn): Boolean =
+      kind(defn).exists(memberKind.contains)
+
+    def matchesSimpleDefinition(rhs: Term): Boolean =
+      rhs.is[Lit] && skipSimpleDefinitions
+
     tree
       .collect {
         case t @ Defn.Val(mods, _, None, body)
-            if t.hasMod(mod"implicit") &&
-              !isImplicitly(body) =>
+            if t.hasMod(mod"implicit") && !isImplicitly(body)
+              || !matchesSimpleDefinition(body)
```

Comment:
Seems like this complex condition is repeated for val/var/def. Can we extract it into a helper? Something like

```scala
def isRewriteCandidate(t: Defn, mods: Seq[Mod], body: Term): Boolean = ???
```